irohad/ametsuchi/impl/postgres_setting_query.cpp

```diff
@@ -0,0 +1,60 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "ametsuchi/impl/postgres_setting_query.hpp"
+
+#include <boost/lexical_cast.hpp>
+#include <interfaces/common_objects/types.hpp>
+#include "logger/logger.hpp"
+
+using namespace iroha::ametsuchi;
+
+namespace {
+  template <typename T>
+  bool getValueFromDb(soci::session &sql,
+                      const shared_model::interface::types::SettingKeyType &key,
+                      T &destination) {
+    boost::optional<shared_model::interface::types::SettingValueType> value;
+
+    sql << "SELECT setting_value FROM setting WHERE setting_key = :key",
```

Comment:
It should be wrapped in try-catch as I understand.

Reply:
Wrapped